### ISSUE TITLE
I can't account update address with password if It was generated without password initially.

### ISSUE THREAD

**Issue by JJNET** [ethereum/go-ethereum#22496]
#### System information

Geth version: 1.9.23-stable
OS & Version: Linux
Commit hash : 8c2f271528f9cccf541c6ea1c022e98407f26872

#### Expected behaviour
1. creating geth new account without password.(0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2)
2. geth attach rpc:http://127.0.0.1:8545
3. personal.unlockAccount(eth.accounts[79]); 
Unlock account 0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2
Passphrase:(Enter only)
true
4. There is no update features with web3.js. So I have to use geth CLI.
5. geth account update 0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2
6. Unlocking account 0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2| Attempt 1/3
Passphrase:
0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2
Account '0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2' unlocked.
Please give a new password. Do not forget this password.
Passphrase: 12345678
Repeat Passphrase: 12345678
0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2
7. finally password updated

#### Actual behaviour
1. creating geth new account without password.(0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2)
2. geth attach rpc:http://127.0.0.1:8545
3. personal.unlockAccount(eth.accounts[79]); 
Unlock account 0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2
Passphrase:(Enter only)
true
4. There is no update features with web3.js. So I have to use geth CLI.
5. geth account update 0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2
6. Unlocking account 0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2| Attempt 1/3
Passphrase:(Enter only)
Fatal: Failed to unlock account 0x2e34a48b0da51f55d569e0af7d0d6836ea8cd5a2 (no key for given address or file)

#### Steps to reproduce the behaviour

I think geth CLI don't permit non-password input unlike Web3.js.
Old geth permit sendTransactions without password. But now it is impossible. 
So I can't send ETH coin  without updating some kind of password for old address.

This is critical bug. Help me!

**Comment by karalabe** [comment]
Works for me:

```
$ geth --datadir=/tmp/ggg --goerli console
> personal.newAccount()
Passphrase: 
Repeat passphrase: 
"0x9f56b0af8691f042fd88526da4a403f533456f45"

$ geth --datadir=/tmp/ggg account update 0x9f56b0af8691f042fd88526da4a403f533456f45
Unlocking account 0x9f56b0af8691f042fd88526da4a403f533456f45 | Attempt 1/3
Password: 
INFO [03-17|10:10:16.819] Unlocked account                         address=0x9f56b0AF8691f042FD88526DA4A403F533456f45
Please give a new password. Do not forget this password.
Password: 
Repeat password:
```

**Comment by karalabe** [comment]
Are you sure you're not using a different datadir for your main Geth instance and when you want to change the password, you don't specify it so it can't find the account?

**Comment by JJNET** [comment]
> Are you sure you're not using a different datadir for your main Geth instance and when you want to change the password, you don't specify it so it can't find the account?

I can't remember datadir have been changed or not. And I don't think the problem is about finding account because web3.js can unlock same account via geth daemon successfully despite that CLI can't accept null password. 
And don't mind anymore. This problem is about testnet. I changed to rinkeby test network from ropsten test network already.

Thank you!
